### PR TITLE
Move contributor documentation into a dedicated folder

### LINKED ISSUE #1922: Document environment variables that can be used to modify swift-syntax’s build or test execution
We should have a single documentation document that describes all the environment variables that can be used to modify the behavior of Package.swift (e.g. `SWIFT_BUILD_SCRIPT_ENVIRONMENT`) and test execution (e.g. `SKIP_LONG_TESTS`)

### PR DESCRIPTION
**Pull request by ahoppen**
We had documentation targeted at swift-syntax contributors spread in the docc documentation of SwiftSyntax and SwiftParser. I think it doesn’t belong there because users of swift-syntax don’t care about it. It also makes it harder to find because you need to look in two places.

Move all that documentation to a single `Contributor Documentation` folder.

---

Also add a new document that describes the environment variables that can be used to modify swift-syntax’s build or test execution.

rdar://112473317
Fixes #1922

### CODE REVIEW

**Comment by ahoppen** [comment]
@swift-ci Please test

File: Contributor Documentation/Environment variables.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+The following environment variables can be used to control some behavior in swift-syntax.
+
+## Build time
+
+- `SWIFT_BUILD_SCRIPT_ENVIRONMENT`: Enable assertions in release builds and remove the dependency of swift-syntax on os_log.
+- `SWIFTCI_USE_LOCAL_DEPS`: Assume that all of SourceKit-LSP’s dependencies are checked out next to it and use those instead of cloning the repositories. Primarily intended for CI environments that check out related branches.
+- `SWIFTPARSER_ENABLE_ALTERNATE_TOKEN_INTROSPECTION`: Mutate the input of `assertParse` test cases. See CONTRIBUTING.md for more information.
+- `SWIFTSYNTAX_BUILD_DYNAMIC_LIBRARIES`: Build all libraries as *dynamic* instead of *automatic*. This allows us to build swift-syntax as dynamic libraries, which in turn allows us to build SourceKit-LSP using SwiftPM on Windows. Linking swift-syntax statically into sourcekit-lsp exceeds the maximum number of exported symbols on Windows.
+- `SWIFTSYNTAX_ENABLE_RAWSYNTAX_VALIDATION`: Check that the layout of the syntax tree is correct when creating or modifying nodes. See CONTRIBUTING.md for more information
```
Comment (Matejkob) [inline Contributor Documentation/Environment variables.md:11]: We could also include a link to the article from the `SwiftSyntax` target that discusses raw syntax validation for macro testing: [Sources/SwiftSyntax/Documentation.docc/RawSyntaxValidation.md](https://github.com/swiftlang/swift-syntax/blob/main/Sources/SwiftSyntax/Documentation.docc/RawSyntaxValidation.md)

**Review by Matejkob — COMMENTED** [review]
I really like this division of the documentation. Moving contributor-specific content to its own folder makes things much clearer! 

The document on environment variables looks great too!

File: Contributor Documentation/Environment variables.md (hunk 1 above)
Comment (ahoppen) [reply Contributor Documentation/Environment variables.md:11]: That document is target for macro developers. I think the information in `CONTRIBUTING.md` match better here.

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Review by bnbarham — APPROVED** [review]

**Pull request merged**